src/main/java/org/disq_bio/disq/HtsjdkReadsRddStorage.java

```diff
@@ -0,0 +1,189 @@
+package org.disq_bio.disq;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.ValidationStringency;
+import htsjdk.samtools.util.Locatable;
+import java.io.IOException;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.disq_bio.disq.impl.file.FileSystemWrapper;
+import org.disq_bio.disq.impl.file.HadoopFileSystemWrapper;
+import org.disq_bio.disq.impl.file.NioFileSystemWrapper;
+import org.disq_bio.disq.impl.formats.sam.AbstractSamSource;
+import org.disq_bio.disq.impl.formats.sam.SamFormat;
+
+/** The entry point for reading or writing a {@link HtsjdkReadsRdd}. */
+public class HtsjdkReadsRddStorage {
+
+  private JavaSparkContext sparkContext;
+  private int splitSize;
+  private ValidationStringency validationStringency = ValidationStringency.DEFAULT_STRINGENCY;
+  private boolean useNio;
+  private String referenceSourcePath;
+
+  /**
+   * Create a {@link HtsjdkReadsRddStorage} from a Spark context object.
+   *
+   * @param sparkContext the Spark context to use
+   * @return a {@link HtsjdkReadsRddStorage}
+   */
+  public static HtsjdkReadsRddStorage makeDefault(JavaSparkContext sparkContext) {
```

Comment:
Similar thing here, perhaps public API methods should accept `SparkContext` instead of `JavaSparkContext`.